aspnetcore/host-and-deploy/iis/iis-http2.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Using HTTP/2 with IIS
+author: rick-anderson
+description: Learn how to use HTTP/2 features with IIS.
+monikerRange: '>= aspnetcore-5.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 01/13/2020
+no-loc: ["ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: host-and-deploy/iis/iis-http2
+---
+# Using HTTP/2 with IIS
+
+By [Justin Kotalik](https://github.com/jkotalik)
+
+[HTTP/2](https://httpwg.org/specs/rfc7540.html) is supported with ASP.NET Core in the following IIS deployment scenarios:
+
+* In-process
+  * Windows Server 2016/Windows 10 or later; IIS 10 or later
+  * TLS 1.2 or later connection
+* Out-of-process
+  * Windows Server 2016/Windows 10 or later; IIS 10 or later
+  * Public-facing edge server connections use HTTP/2, but the reverse proxy connection to the [Kestrel server](xref:fundamentals/servers/kestrel) uses HTTP/1.1.
+  * TLS 1.2 or later connection
+
+For an in-process deployment when an HTTP/2 connection is established, [`HttpRequest.Protocol`](xref:Microsoft.AspNetCore.Http.HttpRequest.Protocol*) reports `HTTP/2`. For an out-of-process deployment when an HTTP/2 connection is established, [`HttpRequest.Protocol`](xref:Microsoft.AspNetCore.Http.HttpRequest.Protocol*) reports `HTTP/1.1`.
+
+For more information on the in-process and out-of-process hosting models, see <xref:host-and-deploy/aspnet-core-module>.
+
+HTTP/2 is enabled by default. Connections fall back to HTTP/1.1 if an HTTP/2 connection isn't established. For more information on HTTP/2 configuration with IIS deployments, see [HTTP/2 on IIS](/iis/get-started/whats-new-in-iis-10/http2-on-iis).
+
+## Advanced HTTP/2 features to support gRPC
+
+Additional HTTP/2 features in IIS support gRPC, including support for response trailers and sending reset frames.
+
+Requirements to run gRPC on IIS:
+
+* Must use in-process hosting.
+* Windows 10, OS Build 20300.1000 or later (may require being on Windows Insider Builds).
+* TLS 1.2 or later connection
+
+### Trailers
```

Comment:
Until we resolve #26509, should we add this to kestrel.md?

@guardrex Is there a way to include content from one markdown file into another to reduce duplication?

Reply:
Yes you can reference another file.

Example:
* https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/includes/gRPCazure.md
* https://github.com/dotnet/AspNetCore.Docs/blame/23bd8d205726bf80904800f6e98a98fee8ea0bfa/aspnetcore/grpc/index.md#L29

Reply:
We should do that that for this. Even if it's temporary, it reduces the risk of the sections becoming out of sync.

Reply:
Place the content into an INCLUDE file and place it into an `iis` folder (node) `includes` folder.

`host-and-deploy/iis/includes/trailers.md` ... with the content.

Then ...

```
[!INCLUDE[](includes/trailers.md)]
```

... in the text of a topic in the `iis` folder.

Reply:
What should I do about DeclareTrailers between these docs? Should I just go above and beyond and describe why it's needed for HTTP/1.1?